planner/core/stats.go

```diff
@@ -192,16 +197,28 @@ func (ds *DataSource) DeriveStats(childStats []*property.StatsInfo, selfSchema *
 		}
 	}
 	ds.stats = ds.deriveStatsByFilter(ds.pushedDownConds, ds.possibleAccessPaths)
+	if tiflashPath != nil {
+		_, err := ds.deriveTablePathStats(tiflashPath, ds.pushedDownConds, false)
+		if err != nil {
+			return nil, err
+		}
+	}
 	for _, path := range ds.possibleAccessPaths {
+		if path.StoreType == kv.TiFlash {
+			continue
+		}
 		if path.IsTablePath {
 			noIntervalRanges, err := ds.deriveTablePathStats(path, ds.pushedDownConds, false)
 			if err != nil {
 				return nil, err
 			}
 			// If we have point or empty range, just remove other possible paths.
-			if noIntervalRanges || len(path.Ranges) == 0 {
+			if (noIntervalRanges || len(path.Ranges) == 0) && path.StoreType == kv.TiKV {
 				ds.possibleAccessPaths[0] = path
 				ds.possibleAccessPaths = ds.possibleAccessPaths[:1]
+				if tiflashPath != nil {
+					ds.possibleAccessPaths = append(ds.possibleAccessPaths, tiflashPath)
```

Comment:
Why should we keep the tiflash path here? would the behavior be inconsistent with other hints like `USE_INDEX`? since those indexes are removed as well ignoring potential hints.

Reply:
I'm afraid it will affect other index hints. Since the path has been filtered.